### PR TITLE
GOV:325 Implementation of de duplication story

### PR DESCRIPTION
**Pull request by danishjamal104**
[GOV-325](https://mifosforge.jira.com/browse/GOV-325)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above.

- [x] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

[GOV-325]: https://mifosforge.jira.com/browse/GOV-325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java [hunk 1]
```diff
@@ -48,11 +47,11 @@ public void setup() {
                 variables.put(ORDERED_BY, exchange.getProperty(ORDERED_BY));
             }
 
-            if (workerConfig.isTransactionDeduplicationEnabled) {
-                List<Transaction> transactionList = exchange.getProperty(TRANSACTION_LIST, List.class);
-                removeDuplicates(transactionList, workerConfig.isOrderingWorkerEnabled);
-                variables.put(TRANSACTION_LIST, transactionList);
-            }
+            // if (workerConfig.isTransactionDeduplicationEnabled) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java:50]: delete the commented lines

File: src/main/java/org/mifos/processor/bulk/camel/routes/DeDuplicationRoute.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                        }
+                    }
+
+                    log.info("Duplicate txn: {} and count: {}", duplicateTransactionList, duplicateTxnCount);
+
+                    exchange.setProperty(DUPLICATE_TRANSACTION_COUNT, duplicateTxnCount);
+                    exchange.setProperty(DUPLICATE_TRANSACTION_LIST, duplicateTransactionList);
+                    exchange.setProperty(ORIGINAL_TRANSACTION_LIST, originalTransactionList);
+
+                    // todo update the existing file , create a new duplicate txn file and upload in s3
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/DeDuplicationRoute.java:63]: finish or remove the todo task

**Review by fynmanoj — CHANGES_REQUESTED** [review]
requested minor changes

File: src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java]: done

File: src/main/java/org/mifos/processor/bulk/camel/routes/DeDuplicationRoute.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/camel/routes/DeDuplicationRoute.java]: I missed the comment, the task is already done.

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**